samples: Bluetooth: Changes to connected CS samples

eriksandgren opened the pull request:
This PR consists of two commits with changes in the connected CS samples.

eriksandgren committed:
samples: Bluetooth: Update params in connected CS sample

This change is made to fix the currently broken CS samples, to prevent
them from just printing error messages like:
`Channel sounding not enough memory to store step data (530 > 512)`.

This change updates the CS config and procedure parameters. The
intention with the updated parameter selection is to use parameters that
result in:
- CS procedures with only a single CS subevent per procedure. This is
  required because the samples are not designed to support > 1 subevent
  per procedure and the samples will break down if > 1 subevent per
  procedure is generated.
- A limited number of CS steps per procedure. This is achieved by
  setting `.channel_map_repetition = 1` and only enabling a limited set
  of channels in the channel map. This is required because the samples
  are designed to transfer all of the step data from one procedure, from
  the reflector to the initiator, using a single GATT write operation
  which is limited to 512 bytes.

Signed-off-by: Erik Sandgren <erik.sandgren@nordicsemi.no>

alwa-nordic commented:
I'm missing some details to understand this PR so I can review it:
 - Why is this the correct thing to do? Are there alternatives?
 - How was the number of channels to disable determined?
 - This is a sample. We need the code to be educational. Let's make it clear whether this is e.g. demonstrating a best practice, or e.g. demonstrating a specific technique, or maybe something done to greatly simplify the sample which one wouldn't do in production. I think the code comment should explain *why* there is code there to limit the number of steps in the procedure.

eriksandgren commented:
> I'm missing some details to understand this PR so I can review it:
> 
> * Why is this the correct thing to do? Are there alternatives?
> * How was the number of channels to disable determined?
> * This is a sample. We need the code to be educational. Let's make it clear whether this is e.g. demonstrating a best practice, or e.g. demonstrating a specific technique, or maybe something done to greatly simplify the sample which one wouldn't do in production. I think the code comment should explain _why_ there is code there to limit the number of steps in the procedure.
Thanks for the feedback! I will try to address your questions by updating the code comment

eriksandgren committed:
samples: Bluetooth: Print that sample does not support > 1 subevent

This commit adds a print statement to both connected_cs_initiator and
connected_cs_reflector which will inform the user that the sample cannot
handle more than 1 CS subevent per CS procedure.

This is preferable over the sample failing silently if a user is
attempting to use such a configuration.

Signed-off-by: Erik Sandgren <erik.sandgren@nordicsemi.no>
